src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -1396,12 +1306,12 @@ internal void ResolveConflict([@NotNull]string docID, [@CanBeNull]IConflictResol
                     {
                         // Do this in a batch so that there are no changes to the document between
                         // localDoc read and remoteDoc read
-                        localDoc = new Document(this, docID);
+                        localDoc = new Document(DefaultCollection, docID);
```

Comment:
Still, if the default collection is deleted DefaultCollection will be returning null.  I think this should probably be a separate error condition for the legacy calls.  None of them are going to work if the default collection is deleted.  I think there should also be a test to ensure that it throws an exception instead of possibly crashing.

Reply:
Will add the test case with scenario when the default collection is deleted during conflict resolver. The expected result should throw an exception instead of possibly crashing. I added the note in this ticket: https://issues.couchbase.com/browse/CBL-3080

Reply:
This ResolveConflict in Database may simply call Collection ResolveConflict after the collection replication/collection configuration are implemented I think....

Reply:
Oh I misread "deleted during conflict resolution."  This definitely sounds like it is deleted during the conflict resolver.  However, the database can have no default collection at any time.  Is the legacy API prepared to handle this?  I think this error should be coordinated with @pasin

Reply:
1. In the collection API, CouchbaseLiteException with NotOpen error code will be thrown. We could also add a custom error message such as "The default collection has been deleted."

2. In the legacy API, the behavior will be the same as when the database is deleted, which is throwing InconsistentStateException.

Reply:
#1 has taken care of in different PR https://github.com/couchbase/couchbase-lite-net/pull/1390/files#diff-bd6880d0a8cd12b58b5f73a6389f6e6aa27c78e47bc346467e114b683ad4a796R676-R688
#2 has taken care of here in this commit https://github.com/couchbase/couchbase-lite-net/pull/1388/commits/e1c1d01b5c7a1b5c3b317f00a5653a694bdbeb11